### ISSUE TITLE
Question: It doesn't handle cropping of sticky headers right now? Do you plan on handling it in future?

### ISSUE THREAD

**Issue by sandeepnagra** [samnissen/watir-screenshot-stitch#26]
Here's an example:

![full-page-image](https://user-images.githubusercontent.com/9598669/37723798-a27b03b8-2d05-11e8-8289-5a850913dba6.png)

**Comment by samnissen** [comment]
tl;dr: use the canvas method to capture these sites. I don't think this is solvable in an efficient manner. Please close when satisfied.

-----

Yeah, I've seen that too and it's an unfortunate side effect of the stitching approach. 

### The DOM
But I'm not sure how I could programmatically detect these with anything approaching 100% certainty in the DOM. If high certainty was possible, DOM-detection would be ideal because it might be possible to implement before image-gathering or at least during. 

### OCR
OCR might work, but would have to be employed after the images were stitched, the header sized calculated, then the images would have to be re-gathered (to capture data originally under the header). In short, I don't see implementing an OCR-based strategy, especially because ...

### A workaround
I couldn't more strongly recommend `base64_canvas`, which does not have these problems.

### Next steps
I'll leave the issue with you to close or comment further about detection strategies.

**Comment by sandeepnagra** [comment]
I downloaded the latest version of html2canvas.js and put it under a folder called vendor in root directory of the project. The I ran the following script as mentioned on you README file but it throws an error.

```
require 'watir-screenshot-stitch'

b = Watir::Browser.new :chrome
b.goto "https://www.wikipedia.com/"
b.screenshot.base64_canvas(b)
```

The browser is launched but it stays there and nothing happens. I see following error after 2 mins.

```
/Users/xyz/.rvm/rubies/ruby-2.4.1/bin/ruby -e $stdout.sync=true;$stderr.sync=true;load($0=ARGV.shift) /Users/xyz/RubymineProjects/test/full_page_screenshots.rb
/Users/xyz/.rvm/gems/ruby-2.4.1/gems/watir-6.8.1/lib/watir/wait.rb:49:in `until': timed out after 120 seconds, waiting for true condition on #<Watir::Browser:0x148ed7f983629afc url="https://www.wikipedia.org/" title="Wikipedia"> (Watir::Wait::TimeoutError)
	from /Users/xyz/.rvm/gems/ruby-2.4.1/gems/watir-6.8.1/lib/watir/wait.rb:125:in `wait_until'
	from /Users/xyz/.rvm/gems/ruby-2.4.1/gems/watir-screenshot-stitch-0.6.0/lib/watir-screenshot-stitch.rb:74:in `base64_canvas'
	from /Users/xyz/RubymineProjects/test/full_page_screenshots.rb:40:in `<top (required)>'
	from -e:1:in `load'
	from -e:1:in `<main>'

Process finished with exit code 1
```

**Comment by sandeepnagra** [comment]
Seems like it could be an issue with html2canvas itself. Is it working for you?

I see the following error in browser console.

```
1ms html2canvas: onrendered option is deprecated, html2canvas returns a Promise with the canvas as the value
  | error | @ | VM26:3058
-- | -- | -- | --
  | html2canvas | @ | VM26:3730
  | genScreenshot | @ | VM28:372
  | (anonymous) | @ | VM28:372
  | callFunction | @ | VM28:361
  | (anonymous) | @ | VM28:371
  | (anonymous) | @ | VM28:372
```

**Comment by sandeepnagra** [comment]
I raised a PR with a fix for this issue - https://github.com/samnissen/watir-screenshot-stitch/pull/27

I ran the following script but the rendering it not that great especially the images on the webpage:

```
require 'watir-screenshot-stitch'

b = Watir::Browser.new :chrome
b.goto "https://advisors.massmutual.com/"

img = b.screenshot.base64_canvas(b)
png = Base64.decode64(img)
File.open('test.png', 'wb') { |f| f.write(png) }
```
Here's how it looks:

![test](https://user-images.githubusercontent.com/9598669/37751496-46669ede-2d69-11e8-8cad-edfdab089d89.png)

I might stick to the previous method.
